src/community/proxy-base-ext/src/main/java/org/geoserver/proxybase/ext/web/WebLinkingManglingRulePage.java

```diff
@@ -0,0 +1,100 @@
+import org.apache.wicket.model.IModel;
+import org.apache.wicket.model.Model;
+import org.geoserver.web.GeoServerSecuredPage;
+
+/** The editing page for the {@link WebLinkingUrlMangler}. */
+public class WebLinkingManglingRulePage extends GeoServerSecuredPage {
+    public WebLinkingManglingRulePage(Optional<RuleModel> optionalRuleModel) {
+        CompoundPropertyModel<RuleModel> simpleRuleModel =
+                new CompoundPropertyModel<>(optionalRuleModel.orElse(new RuleModel()));
+        Form<RuleModel> form = new Form<>("form");
+        add(form);
+        List<WrappedTab> tabs = new ArrayList<>();
+        if (!optionalRuleModel.isPresent() || optionalRuleModel.get().getPosition() != null) {
+            tabs.add(
+                    new WrappedTab("Proxy Base Extension Rule", simpleRuleModel) {
+                        @Override
+                        public Panel getPanel(String panelId) {
+                            return new SimpleRulePanel(panelId, simpleRuleModel);
+                        }
+                    });
+        }
+        AjaxTabbedPanel tabbedPanel = new AjaxTabbedPanel<>("tabs", tabs);
+        form.add(tabbedPanel);
+        form.add(
+                new SubmitLink("save") {
+                    @Override
+                    public void onSubmit() {
+                        try {
+                            WrappedTab selectedTab = tabs.get(tabbedPanel.getSelectedTab());
+                            RuleModel ruleModel = selectedTab.getModel().getObject();
+                            RulesModel.saveOrUpdate(ruleModel);
+                            doReturn(WebLinkingManglingConfigPage.class);
+                        } catch (Exception exception) {
+                            error(exception);
+                        }
+                    }
+                });
+        form.add(new BookmarkablePageLink<>("cancel", WebLinkingManglingConfigPage.class));
+    }
+
+    /** A tab that wraps a panel. */
+    public abstract class WrappedTab extends AbstractTab {
+
+        private final IModel<RuleModel> model;
+
+        /**
+         * Constructor.
+         *
+         * @param title the title
+         * @param model the model
+         */
+        public WrappedTab(String title, IModel<RuleModel> model) {
+            super(new Model<>(title));
+            this.model = model;
+        }
+
+        /**
+         * Returns the model.
+         *
+         * @return the model
+         */
+        public IModel<RuleModel> getModel() {
+            return model;
+        }
+    }
+
+    /** A simple rule panel. */
+    public class SimpleRulePanel extends Panel {
+        /**
+         * Constructor.
+         *
+         * @param panelId the panel id
+         * @param model the model
+         */
+        public SimpleRulePanel(String panelId, IModel<RuleModel> model) {
+            super(panelId, model);
+            add(new NumberTextField<Integer>("position").setMinimum(1).setRequired(true));
```

Comment:
When creating a new rule the default is "zero":

![image](https://user-images.githubusercontent.com/325433/233657616-3ada786a-6db9-428a-91f8-9eede5e70a35.png)

Actually not sure why the minimum has to be 1, but either make the minimum zero, or make sure the default value is sensible.

Reply:
done